app/services/hyrax/workflow/workflow_action_service.rb

```diff
@@ -20,7 +20,7 @@ def run
         comment = create_sipity_comment
         handle_sipity_notifications(comment: comment)
         handle_additional_sipity_workflow_action_processing(comment: comment)
-        subject.work.update_index # So that the new actions and state are written into solr.
+        subject.work.try(:update_index) # So that the new actions and state are written into solr.
```

Comment:
this is a hard `ActiveFedora` dependency. i wasn't ready to chase it down, but i think we should extract it.

one issue for me taking this on is that, where i am in this process right now, i don't know what `WorkflowActionService` *does* or why it wants to reindex the model.

Reply:
cc: @jeremyf

Reply:
Ugh, not well documented.  I'm at the end of my evening, but may have some insights.

Sipity has the concept of callbacks.  When you take an action it may fire functions.  Those functions can do _anything_; Which would be why we might need to update the index.

(That's the proverbial straw I have at this point)

Reply:
At this point, I'm at 🍻 o'clock.  It would be reasonable to move the `subject.update_index` into the `ActionTakenService`.  After all, if the ActionTakenService does something, then _it_ should be the one to update the index.

I think from a responsibility stand-point if I generate a comment, I shouldn't need to update the index. (But are we storing comments in the index and rendering them there?)

Reply:
oh, that makes sense.

we need this index updated because we need this index to be real time, and `handle_additional_sipity_workflow_action_processing` might have changed metadata or other indexed (non-sipity) values?

is your proposal essentially that, here, we plead ignorance to all of that and expect actions to handle it?

Reply:
notably, https://github.com/samvera/hyrax/blob/130c4e600318a9194725b35dd0fd6e19e5108dd9/app/services/hyrax/workflow/action_taken_service.rb#L35 already saves the target, which needs to be refactored for valkyrie itself, but also should be reindexing.

wondering if this call is already unneeded?

Reply:
That certainly looks like we need the call; The issue would be "Are we expecting comments in the SOLR document?"

Reply:
seems pretty weird. it seems like it would be preferable if the workflows didn't know anything about object persistence or indexing

Reply:
It's an odd thing.  A state change of an object might impact persistence and indexing; However, I think this should be uni-directional.  That is to say, a state change can write to persistence via some semblance of a command.

It highlights that we have traditionally overloaded the index with TOO much data for the variety of purposes.